Prox/Prox/PlaceDetails/FilterViewController.swift

```diff
@@ -0,0 +1,193 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+protocol FilterViewControllerDelegate: class {
+    func filterViewController(_ filterViewController: FilterViewController, didUpdateFilters filters: [PlaceFilter])
+    func filterViewController(_ filterViewController: FilterViewController, didDismissWithFilters filters: [PlaceFilter])
+}
+
+/// A drop-down view from the top of the screen that displays a list of categories to filter.
+class FilterViewController: UIViewController {
+    weak var delegate: FilterViewControllerDelegate?
+
+    fileprivate var showFilterConstraints = [NSLayoutConstraint]()
+    fileprivate var hideFilterConstraints = [NSLayoutConstraint]()
+
+    fileprivate let background = UIButton()
+    private let stackView = UIStackView()
+    private let filters: [PlaceFilter]
+    private let placeCountLabel = UILabel()
+
+    init(filters: [PlaceFilter]) {
+        self.filters = filters
+        super.init(nibName: nil, bundle: nil)
+
+        modalPresentationStyle = .overCurrentContext
+        transitioningDelegate = self
+
+        for (i, filter) in filters.enumerated() {
+            let button = FilterButton()
+            button.setTitle(filter.label, for: .normal)
+            button.isSelected = filter.enabled
+            button.tag = i
+            button.addTarget(self, action: #selector(didToggleFilter(sender:)), for: .touchUpInside)
+            stackView.addArrangedSubview(button)
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        background.backgroundColor = Colors.filterOverlayDim
+        background.alpha = 0
+        background.addTarget(self, action: #selector(didPressClose), for: .touchDown)
+        view.addSubview(background)
+
+        let container = UIView()
+        container.backgroundColor = .white
+        view.addSubview(container)
+
+        placeCountLabel.font = UIFont.systemFont(ofSize: 20)
```

Comment:
nit: we have a Fonts file for defining these (but I'm not convinced it's particularly useful).